packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx

```diff
@@ -174,304 +193,354 @@ const ResourceViewActionMenu: FC<ResourceViewActionMenuProps> = ({
-                                                >
-                                                    {space.name}
-                                                </Menu.Item>
-                                            ),
-                                        )}
-                                    </Fragment>
-                                ))}
-
-                                <Can
-                                    I="create"
-                                    this={subject('Space', {
-                                        organizationUuid:
-                                            user.data?.organizationUuid,
-                                        projectUuid,
-                                    })}
-                                >
-                                    {spaces.length > 0 ? (
-                                        <Menu.Divider />
-                                    ) : null}
+                                Add to Dashboard
+                            </Menu.Item>
+                        )}
+                    {userCanPromoteChart &&
+                        item.type !== ResourceViewItemType.SPACE && (
+                            <Tooltip
+                                label="You must enable first an upstream project in settings > Data ops"
+                                disabled={
+                                    project?.upstreamProjectUuid !== undefined
+                                }
+                                withinPortal
+                            >
+                                <div>
                                     <Menu.Item
-                                        component="button"
-                                        role="menuitem"
+                                        disabled={
+                                            project?.upstreamProjectUuid ===
+                                            undefined
+                                        }
                                         icon={
                                             <MantineIcon
-                                                icon={IconPlus}
-                                                size={18}
+                                                icon={IconDatabaseExport}
                                             />
                                         }
                                         onClick={() => {
-                                            onAction({
-                                                type: ResourceViewItemAction.CREATE_SPACE,
-                                                item,
-                                            });
+                                            if (
+                                                item.type ===
+                                                ResourceViewItemType.CHART
+                                            ) {
+                                                getPromoteChartDiff(
```

Comment:
I changed this: 
instead of promoting, I'm calling getPromoteChart/Dashboard diff to show the diff on the modal. The promotion happens on the modal now